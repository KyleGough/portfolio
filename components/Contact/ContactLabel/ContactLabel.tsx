import { clsx } from 'clsx';
import React from 'react';

interface ContactLabelProps {
  valid: boolean;
  isSent: boolean;
  htmlFor: string;
  children: React.ReactNode;
}

export const ContactLabel: React.FC<ContactLabelProps> = ({
  valid,
  isSent,
  htmlFor,
  children,
}) => (
  <label
    className={clsx(
      { 'text-link-hover': isSent },
      { 'text-link group-focus-within:text-link-hover': valid && !isSent },
      { 'text-error': !valid && !isSent },
      'transition-colours',
      'duration-200',
      'text-lg',
      'mt-8',
      'mb-2',
      'mr-2'
    )}
    htmlFor={htmlFor}
  >
    {children}
  </label>
);
