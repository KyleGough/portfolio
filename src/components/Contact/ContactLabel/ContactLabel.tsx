import React from 'react';

interface ContactLabelProps {
  valid: boolean;
  htmlFor: string;
  children: React.ReactNode;
}

export const ContactLabel: React.FC<ContactLabelProps> = ({
  valid,
  htmlFor,
  children,
}) => (
  <label
    className={`${
      valid ? 'text-link' : 'text-error'
    } transition-colours duration-200 text-lg mt-8 mb-2 mr-2`}
    htmlFor={htmlFor}
  >
    {children}
  </label>
);
