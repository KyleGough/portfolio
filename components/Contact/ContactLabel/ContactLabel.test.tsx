import { render, screen } from '@testing-library/react';
import React from 'react';

import { ContactLabel } from './ContactLabel';

describe('ContactLabel component', () => {
  it('renders', () => {
    const { container } = render(
      <ContactLabel isSent={true} valid={true} htmlFor="email">
        Email
      </ContactLabel>
    );
    expect(container).toMatchSnapshot();
  });

  it('component has link colours if valid', () => {
    render(
      <ContactLabel isSent={false} valid={true} htmlFor="email">
        Email
      </ContactLabel>
    );

    const label = screen.getByText('Email');
    expect(label).toBeVisible();
    expect(label.className.includes('text-link')).toBe(true);
  });

  it('component has error colours if invalid', () => {
    render(
      <ContactLabel isSent={false} valid={false} htmlFor="name">
        Name
      </ContactLabel>
    );

    const label = screen.getByText('Name');
    expect(label).toBeVisible();
    expect(label.className.includes('text-error')).toBe(true);
  });
});
