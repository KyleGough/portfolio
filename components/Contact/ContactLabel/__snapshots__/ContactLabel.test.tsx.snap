// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactLabel component renders 1`] = `
<div>
  <label
    class="text-link group-focus-within:text-link-hover transition-colours duration-200 text-lg mt-8 mb-2 mr-2"
    for="email"
  >
    Email
  </label>
</div>
`;
